### ISSUE TITLE
`// swiftlint:disable identifier_name` does not work with Swift 4 toolchain

### ISSUE THREAD

**Issue by norio-nomura** [realm/SwiftLint#1741]
I'm using `// swiftlint:disable identifier_name` at [Yams/Tests/YamsTests/EncoderTests.swift:15](https://github.com/jpsim/Yams/pull/43/files#diff-73a608f5dae28b72f46bae980aafaea1R15).
But swiftlint reports violations of `identifier_name` on [travis job](https://travis-ci.org/jpsim/Yams/jobs/260432361#L259).
This issue may occur locally with swiftlint 0.21.0, but I can not identify the condition.
It seems putting `// swiftlint:disable:this identifier_name` at each violated lines works.

**Comment by jpsim** [comment]
Interesting finding... Looks like this only occurs with a Swift 4 toolchain. Running with a Swift 3 toolchain picks up the multi-rule command just fine.

**Comment by norio-nomura** [comment]
It looks `--no-cache` affects to result.

**Comment by jpsim** [comment]
> It looks `--no-cache` affects to results.

Only if with/without the cache, different Swift toolchains are used.

**Comment by norio-nomura** [comment]
You are right.
In my test, the result of Xcode 9 beta 4 was cached and `-no-cache` used Xcode 8.3.3.

**Comment by marcelofabri** [comment]
Related: https://github.com/realm/SwiftLint/issues/1709

**Comment by marcelofabri** [comment]
Maybe the issue is that there's a comma in the command?

```swift
// swiftlint:disable identifier_name, line_length
```

The correct syntax, according to [README](https://github.com/realm/SwiftLint#disable-rules-in-code), would be:

```swift
// swiftlint:disable identifier_name line_length
```

**Comment by jpsim** [comment]
Yes, rule identifiers need to be separated by whitespace: https://github.com/realm/SwiftLint/blob/0.21.0/Source/SwiftLintFramework/Models/Command.swift#L81-L83

Although we could allow commas as separators too.

**Comment by norio-nomura** [comment]
Oh, I did not know the correct syntax.
The behavior difference by toolchain was caused by `#if swift (> = 4.0)`.
Apparently it seems to be the correct behavior.
Thanks! 🙏
